Projectoid.js

```diff
@@ -0,0 +1,313 @@
+ */
+function apiCall(options) {
+  if (!options) {
+    throw libPrefix + ": apiCall: options not found";
+  }
+  let url = API_URL + "/" + options.path;
+  if (options.method.toLowerCase() === "get") {
+    url += "?" + jTQS(options.body);
+  }
+  let apiData = {
+    url: url,
+    headers: {
+      ...(options.method.toLowerCase() === "get"
+        ? {}
+        : { "Content-Type": "application/json" }),
+    },
+    body: options.body,
+    folow_redirects: true,
+    success: libPrefix + "onApiAnswer " + options.onSuccess,
+    error: libPrefix + "onApiError",
+  };
+
+  // Check the method and use the corresponding HTTP function
+  if (options.method.toLowerCase() === "get") {
+    HTTP.get(apiData);
+  } else if (options.method.toLowerCase() === "post") {
+    HTTP.post(apiData);
+  } else {
+    throw "Unsupported HTTP method: " + options.method;
+  }
+}
+
+/**
+ * saveToken() - Saves the Projectoid access token.
+ *
+ * @since 1.0.0
+ *
+ * @param {string} access_token - The Projectoid access token to save.
+ */
+function saveToken(access_token) {
+  if (!access_token) {
+    throw libPrefix + ": Projectoid Access Token Not Found";
+  }
+  if (access_token.length != 32) {
+    throw libPrefix + ": Projectoid Access Token is Wrong";
+  }
+  Bot.setProperty("Projectoid_AccessToken", access_token);
+}
+
+/**
+ * addChat() - Adds a chat to Projectoid.
+ *
+ * @since 1.0.0
+ *
+ * @param {number} chatid - The chat ID to add.
+ * @param {string} access_token - The Projectoid access token.
+ * @param {string} command - The command to execute on success.
+ */
+function addChat(chatid, access_token, command) {
+  if (isNaN(parseInt(chatid))) {
+    throw libPrefix + ": addChat: incorrect chat id";
+  }
+
+  if (!access_token) {
+    var access_token = Bot.getProperty("Projectoid_AccessToken");
+    if (!access_token) {
+      throw libPrefix + ": addChat: Access Token Not Found";
+    }
+  }
+
+  let requestData = {
+    method: "post",
+    path: "telegram/botpanel/adduser.php",
+    body: {
+      bot_id: bot.token.split(":")[0],
+      user_id: chatid,
+      access_token,
+    },
+    onSuccess: !command ? null : command,
+  };
+  apiCall(requestData);
+}
+
+//---------------------Broadcast Functions Starts----------------------------
+/**
+ * initiateBroadcast() // initiateForwardBroadcast() // initiateCopyBroadcast() - Broadcasts a message to multiple chats.
+ *
+ * @since 1.0.0
+ *
+ * @param {object} options - The broadcast options.
+ */
+function initiateBroadcast(options) {
+  if (!options) {
+    throw libPrefix + ": initiateBroadcast: options not found";
+  }
+  if (!options.method) {
+    throw libPrefix + ": initiateBroadcast: broadcast method not found";
+  }
+  if (options.method == "forwardMessage") {
+    initiateForwardBroadcast(options);
+    return;
+  }
+  if (options.method == "copyMessage") {
+    initiateCopyBroadcast(options);
+    return;
+  }
+
+  let text = !options.text ? null : options.text;
```

Comment:
const  text = options.text - it is enough ?

Reply:
no (sometimes) - to prevent the errors saying `undefined "options.something"`